src/app/settings/admin/constraint-templates/constraint-templates-data-dialog/component.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, Inject, OnDestroy, OnInit} from '@angular/core';
+import {FormControl, FormGroup, Validators} from '@angular/forms';
+import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material/dialog';
+import {ConstraintTemplate} from '@shared/entity/opa';
+import {dump, load} from 'js-yaml';
+import * as _ from 'lodash';
+import {Subject} from 'rxjs';
+
+export interface ConstraintTemplateDataDialogConfig {
+  title: string;
+  confirmLabel: string;
+  isEditing: boolean;
+
+  // Constraint Template has to be specified only if dialog is used in the edit mode.
+  constraintTemplate?: ConstraintTemplate;
+}
+
+export enum Controls {
+  TemplateName = 'name',
+  Spec = 'spec',
+}
+
+@Component({
+  selector: 'km-add-constraint-template-dialog',
+  templateUrl: './template.html',
+  styleUrls: ['./style.scss'],
+})
+export class ConstraintTemplatesDataDialogComponent implements OnInit, OnDestroy {
```

Comment:
I'd follow the selector naming `AddConstraintTemplateDialogComponent`. I guess we could actually drop the `Component` suffix for the dialogs since the dialog is like a special component already.

Reply:
Should I also name it `AddConstraintTemplateDialog` if it is used for Add and Edit? 🤔

Reply:
Then this is totally confusing since the selector is named `km-add-constraint-template-dialog`. If there are multiple modes handled by this dialog then simply drop Add/Edit prefix from the name.

Reply:
oh you're right. I wasn't aware of the `add` in the selector. not sure what happened there 🤔 `km-constraint-template-dialog` sounds good to me?!